PrivateClinicsWebNet.Infrastructure/AppointmentInfrastructure/Repositories/AppointmentRepository.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.EntityFrameworkCore;
+using PrivateClinicsWebNet.BusinessLogic.Abstractions;
+using PrivateClinicsWebNet.BusinessLogic.Entities;
+using PrivateClinicsWebNet.DataAccess;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Abstractions;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.DTOs;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Exceptions;
+using PrivateClinicsWebNet.Infrastructure.Shared.Wrapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Repositories
+{
+    public class AppointmentRepository(ApplicationDbContext _context) : IAppointmentRepository
+    {
+        private enum AppointmentsSupportEntities
+        {
+            Doctor,
+            Patient
+        }
+
+        public async Task<AppointmentResponseDto> GetAppointmentByIdAsync(string id)
+        {
+            var response = await _context.Appointments
+                .Where(x => x.Id.ToString() == id)
+                .Select(e => new AppointmentResponseDto
+                {
+                    AppointmentId = e.Id.ToString(),
+                    PatientName = e.Patient.UserName,
+                    DoctorName = e.Doctor.UserName,
+                    DoctorType = e.Doctor.DoctorType,
+                    Date = e.Date,
+                })
+                .FirstOrDefaultAsync();
+            return response;
+        }
+
+        public async Task<bool> СreateAppointmentAsync(Appointment appointment)
+        {
+            var result = await _context.Appointments
+                .AddAsync(appointment);
+            await _context.SaveChangesAsync();
+            return true;
+        }
+
+        public async Task UpdateAppointmentAsync(Appointment appointment)
+        {
+            var existingAppointment = await _context.Appointments.FindAsync(appointment.Id);
+            if (appointment.DoctorId != null)
+                existingAppointment.DoctorId = appointment.DoctorId;
+            if (appointment.PatientId != null)
+                existingAppointment.PatientId = appointment.PatientId;
```

Comment:
How is that possible for appointment to change the doctor or patient? It's forbidden by business logic I suppose